x-pack/platform/test/onechat_functional/tests/tools/create_tool.ts

```diff
@@ -0,0 +1,86 @@
+  const { common } = getPageObjects(['common']);
+  const testSubjects = getService('testSubjects');
+  const monacoEditor = getService('monacoEditor');
+  const es = getService('es');
+
+  describe('create tool', function () {
+    let testIndexName: string;
+
+    before(async () => {
+      testIndexName = `ftr_onechat_${Date.now()}`;
+      await es.indices.create({ index: testIndexName });
+      await es.index({
+        index: testIndexName,
+        document: { message: 'hello world', numeric: 1, '@timestamp': new Date().toISOString() },
+      });
+      await es.index({
+        index: testIndexName,
+        document: { message: 'second doc', numeric: 2, '@timestamp': new Date().toISOString() },
+      });
+      await es.indices.refresh({ index: testIndexName });
+    });
+
+    after(async () => {
+      try {
+        await es.indices.delete({ index: testIndexName });
+      } catch (e) {
+        // ignore
+      }
+    });
+
+    it('should create an esql tool', async () => {
+      const toolId = `ftr.esql.${Date.now()}`;
+      await common.navigateToApp(AGENT_BUILDER_APP_ID, { path: 'tools/new' });
+      await testSubjects.existOrFail('agentBuilderToolFormPage');
+      await testSubjects.setValue('agentBuilderToolIdInput', toolId);
+
+      await testSubjects.existOrFail('agentBuilderToolTypeSelect');
+      await testSubjects.selectValue('agentBuilderToolTypeSelect', 'esql');
+
+      await testSubjects.setValue('euiMarkdownEditorTextArea', 'FTR created ES|QL tool');
+
+      await testSubjects.existOrFail('agentBuilderEsqlEditor');
+      await monacoEditor.setCodeEditorValue('FROM .kibana | LIMIT 1');
+
+      await testSubjects.click('toolFormSaveButton');
+      await testSubjects.click('toastCloseButton');
+
+      await testSubjects.existOrFail(`agentBuilderToolsTableRow-${toolId}`);
+    });
+
+    it('should create an index search tool', async () => {
+      const toolId = `ftr.index.${Date.now()}`;
+      await common.navigateToApp(AGENT_BUILDER_APP_ID, { path: 'tools/new' });
+      await testSubjects.existOrFail('agentBuilderToolFormPage');
+      await testSubjects.setValue('agentBuilderToolIdInput', toolId);
+
+      await testSubjects.existOrFail('agentBuilderToolTypeSelect');
+      await testSubjects.selectValue('agentBuilderToolTypeSelect', 'index_search');
+
+      await testSubjects.existOrFail('onechatIndexPatternInput');
+      await testSubjects.setValue('onechatIndexPatternInput', testIndexName);
+
+      const descriptionEditor = await testSubjects.find('agentBuilderToolDescriptionEditor');
+      const descriptionTextarea = await descriptionEditor.findByCssSelector(
+        'textarea.euiMarkdownEditorTextArea'
+      );
+      await descriptionTextarea.type('FTR created Index Search tool');
+
+      await testSubjects.click('toolFormSaveButton');
+      await testSubjects.click('toastCloseButton');
+
+      await testSubjects.existOrFail(`agentBuilderToolsTableRow-${toolId}`);
+    });
```

Comment:
We have this onechat page object now https://github.com/elastic/kibana/blob/main/x-pack/platform/test/functional/page_objects/onechat_page.ts. It would be a good idea to start to factorise and encapsulate some of the reusable tool logic in this file - it will also help with the duplication in some of the test suites here 🙏🏽